### ISSUE TITLE
Storing an atomics value in a compressed ETS table allows the atomics to be garbage collected.

### ISSUE THREAD

**Issue by kjnilsson** [erlang/otp#7444]
Storing an atomics value in an ETS table with the `compressed` allows the atomics value to be garbage collected (I assume).

**To Reproduce**

```Eshell V13.2.2  (abort with ^G)
1> R = ets:new(banana, [set, compressed]).
#Ref<0.3977464636.1155137541.146175>
2> ets:insert(R, {1, counters:new(2, [])}).
true
3> [{_, C}] = ets:lookup(R, 1).
[{1,{atomics,#Ref<0.3977464636.1155137541.146182>}}]
4> C.
{atomics,#Ref<0.3977464636.1155137541.146182>}
5> counters:info(C).
** exception error: bad argument
     in function  counters:info/1
        called as counters:info({atomics,#Ref<0.3977464636.1155137541.146182>})
        *** argument 1: invalid atomics reference
```

**Expected behavior**
The expectation would be that it would work the same as if the the ETS table was created _without_ the `compressed` option.

**Affected versions**
tested on 25.3.2 and 26.0.1

**Additional context**
Whether this is a bug or just a gotcha that needs documenting I have no strong opinion on. :)

**Comment by sverker** [comment]
@kjnilsson Thanks for the report.
This is a bug. The compressed ETS format does not keep the referred object alive as it should, in this case the counter array. It will also apply to NIF resource objects.
The symptom will be like reported. The referred object garbage collected prematurely and the reference appears stale, not referring to anything.

**Comment by sverker** [comment]
Proposed fix in #7458.

**Comment by sverker** [comment]
Fix merged to maint branch for OTP 26.1 and also put in queue for patch releases on 24 and 25.
